File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/display/BarChartDisplay.java
package io.github.zwieback.familyfinance.business.chart.display;

import android.os.Parcel;
import android.support.annotation.NonNull;

import io.github.zwieback.familyfinance.business.chart.display.type.BarChartGroupType;

public class BarChartDisplay extends ChartDisplay {

    public static final String BAR_CHART_DISPLAY = "barChartDisplay";

    public static final Creator<BarChartDisplay> CREATOR = new Creator<BarChartDisplay>() {

        @Override
        public BarChartDisplay createFromParcel(Parcel in) {
            return new BarChartDisplay(in);
        }

        @Override
        public BarChartDisplay[] newArray(int size) {
            return new BarChartDisplay[size];
        }
    };

    /**
     * Initialized in the {@link #init} method.
     */
    @SuppressWarnings("NullableProblems")
    @NonNull
    private BarChartGroupType groupType;

    public BarChartDisplay() {
        super();
    }

    public BarChartDisplay(BarChartDisplay display) {
        super(display);
        groupType = display.groupType;
    }

    private BarChartDisplay(Parcel in) {
        super(in);
    }

    @Override
    protected void init() {
        groupType = BarChartGroupType.DAYS;
    }

    @Override
    protected void readFromParcel(Parcel in) {
        groupType = BarChartGroupType.valueOf(in.readString());
    }

    @Override
    public void writeToParcel(Parcel out, int flags) {
        out.writeString(groupType.name());
    }

    @NonNull
    public BarChartGroupType getGroupType() {
        return groupType;
    }

    public void setGroupType(@NonNull BarChartGroupType groupType) {
        this.groupType = groupType;
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/display/BarChartDisplay.kt
package io.github.zwieback.familyfinance.business.chart.display

import io.github.zwieback.familyfinance.business.chart.display.type.BarChartGroupType
import kotlinx.parcelize.Parcelize

@Parcelize
data class BarChartDisplay(
    var groupType: BarChartGroupType = BarChartGroupType.DAYS,
    var isViewIncomeValues: Boolean = true,
    var isViewExpenseValues: Boolean = true,
    var isViewProfitValues: Boolean = false,
    var isViewIncomes: Boolean = true,
    var isViewExpenses: Boolean = true,
    var isViewProfits: Boolean = false,
    var isIncludeTransfers: Boolean = false
) : ChartDisplay<BarChartDisplay>() {

    override fun needRefreshData(newDisplay: BarChartDisplay): Boolean {
        return groupType !== newDisplay.groupType
                || isViewIncomes != newDisplay.isViewIncomes
                || isViewExpenses != newDisplay.isViewExpenses
                || isViewProfits != newDisplay.isViewProfits
                || isIncludeTransfers != newDisplay.isIncludeTransfers
    }

    companion object {
        const val BAR_CHART_DISPLAY = "barChartDisplay"
    }
}
